### DISCUSSION TITLE
Is a .proto package declaration a requirement?

### DISCUSSION THREAD

**Discussion by CorinJG** [tokio-rs/prost#722] [Q&A] [answered]
In the [example](https://docs.rs/prost-build/latest/prost_build/) in the docs, the `*.rs` file generated during the build has a name which matches the package declaration given in the `.proto` file. 

i.e. `package snazzy.items;` is declared in the `.proto` file, and `lib.rs` contains:
```rust
pub mod items {
    include!(concat!(env!("OUT_DIR"), "/snazzy.items.rs"));
}
```
I suspect therefore you _must_ provide a package declaration in every `.proto` file when using Prost. Is this the case?

Just asking as my schema is already in use in other languages without any package declarations in `.proto` files, and adding them can affect things like namespaces and packages of generated code in other languages requiring a few adjustments in client code I'd prefer to avoid but not the end of the world. Thanks

**Comment by LucioFranco** [comment]
I believe its required last I checked. It should be easy to test though :D

**Reply by CorinJG** [reply]
I suspect what prost_build does is use the .proto `package foo;` declarations as the respective target filenames for the emitted bindings. Totally reasonable, just means introducing package declarations into all existing .proto files in the protocol, and since they `import "*.prost"` each other this means modifying the namespaces of the external message types across all files etc, just wondered if there was an alternative but I've done it now anyway:)

**Comment by CorinJG** [comment, marked as answer]
Just found [this](https://docs.rs/prost-build/latest/src/prost_build/lib.rs.html#695)
